temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -394,73 +398,108 @@ public void handle(WorkflowTask task) throws Exception {
           PollWorkflowTaskQueueResponse currentTask = nextWFTResponse.get();
           nextWFTResponse = Optional.empty();
           WorkflowTaskHandler.Result result = handleTask(currentTask, workflowTypeScope);
+          WorkflowTaskFailedCause taskFailedCause = null;
           try {
             RespondWorkflowTaskCompletedRequest taskCompleted = result.getTaskCompleted();
             RespondWorkflowTaskFailedRequest taskFailed = result.getTaskFailed();
             RespondQueryTaskCompletedRequest queryCompleted = result.getQueryCompleted();
 
-            if (taskCompleted != null) {
-              RespondWorkflowTaskCompletedRequest.Builder requestBuilder =
-                  taskCompleted.toBuilder();
-              try (EagerActivitySlotsReservation activitySlotsReservation =
-                  new EagerActivitySlotsReservation(eagerActivityDispatcher)) {
-                activitySlotsReservation.applyToRequest(requestBuilder);
-                RespondWorkflowTaskCompletedResponse response =
-                    sendTaskCompleted(
-                        currentTask.getTaskToken(),
-                        requestBuilder,
-                        result.getRequestRetryOptions(),
-                        workflowTypeScope);
-                // If we were processing a speculative WFT the server may instruct us that the task
-                // was dropped by resting out event ID.
-                long resetEventId = response.getResetHistoryEventId();
-                if (resetEventId != 0) {
-                  result.getResetEventIdHandle().apply(resetEventId);
-                }
-                nextWFTResponse =
-                    response.hasWorkflowTask()
-                        ? Optional.of(response.getWorkflowTask())
-                        : Optional.empty();
-                // TODO we don't have to do this under the runId lock
-                activitySlotsReservation.handleResponse(response);
-              }
-            } else if (taskFailed != null) {
-              sendTaskFailed(
-                  currentTask.getTaskToken(),
-                  taskFailed.toBuilder(),
-                  result.getRequestRetryOptions(),
-                  workflowTypeScope);
-            } else if (queryCompleted != null) {
+            if (queryCompleted != null) {
               sendDirectQueryCompletedResponse(
```

Comment:
The direct query response can also fail due to gRPC message size too large so I think that needs to be under the try as well no?